youtube_dl/utils.py

```diff
@@ -1150,7 +1150,21 @@ def save(self, filename=None, ignore_discard=False, ignore_expires=False):
         compat_cookiejar.MozillaCookieJar.save(self, filename, ignore_discard, ignore_expires)
 
     def load(self, filename=None, ignore_discard=False, ignore_expires=False):
-        compat_cookiejar.MozillaCookieJar.load(self, filename, ignore_discard, ignore_expires)
+        """Load cookies from a file."""
+        if filename is None:
+            if self.filename is not None:
+                filename = self.filename
+            else:
+                raise ValueError(compat_cookiejar.MISSING_FILENAME_TEXT)
+
+        cf = io.StringIO()
+        with open(filename) as f:
+            for line in f:
+                if line.startswith('#HttpOnly_'):
+                    line = line[10:]
```

Comment:
This is error prone, `len` of the actual string should be used instead.

Reply:
i don't see the problem here, the length of `#HttpOnly_`  is constant and won't change, besides the test is there to make sure that breakage in this part of code will be detected.

Reply:
>besides the test is there to make sure that breakage in this part of code will be detected

That's a doubtful argument considering broken core tests at the beginning of this PR.

The length of this string is const until one decides to refactor here something. Using 10 is a variation of code duplication since the length is already implicitly defined in the string literal itself. Also using 10 indicates no relation to the string literal so that one unfamiliar with code who decides to refactor it may forgot to change the number and may be unaware of the tests at all.

Reply:
ok, changed the code to use `len`.